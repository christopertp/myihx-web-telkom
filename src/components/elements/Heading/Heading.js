import React from 'react';
import PropTypes from 'prop-types';
import styles from './styles.scoped.css';

export default function Heading(props) {
  const { children, className } = props;
  const customClass = [styles.root, className].filter(Boolean).join(' ');

  return <h3 className={customClass}>{children}</h3>;
}

Heading.defaultProps = {
  children: null,
  className: '',
};

Heading.propTypes = {
  children: PropTypes.node,
  className: PropTypes.string,
};
