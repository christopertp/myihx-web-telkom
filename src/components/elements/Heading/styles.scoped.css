.root {
  color: #66686a;
  font-size: 0.95rem;
  font-weight: 400;
  letter-spacing: 0.0625rem;
  text-transform: uppercase;
}

@media only screen and (max-width: 48em) {
  .root {
    font-size: 0.75rem;
  }
}
