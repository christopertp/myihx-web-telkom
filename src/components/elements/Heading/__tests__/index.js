import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import Heading from '../Heading';

describe('src/components/elements/Heading', () => {
  test('render', () => {
    const shallow = new ShallowRenderer();
    const tree = shallow.render(<Heading>Text</Heading>);
    expect(tree).toMatchSnapshot();
  });
});
