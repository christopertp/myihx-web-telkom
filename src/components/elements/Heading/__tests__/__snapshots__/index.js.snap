// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/components/elements/Heading render 1`] = `
<h3
  className=""
>
  Text
</h3>
`;
